README.md

```diff
@@ -4,196 +4,307 @@
-
-If the uploader program has more than one hole and there exist dependencies
-between those holes. The program must not fail if there isn't any
-`query_results` to render. This is common in more complicated metadata query
-models and the uploader must handle this.
-
-When the uploader program is ready for a value to be selected. The uploader
-program must set the `value` attribute in the respective `MetaObj` to the
-`valueField` attribute of the object selected. The uploader must then call
-`update_parents()` with the `MetaObj` selected. This will update the `value`
-attribute for many `MetaObj` objects and the updated `MetaUpdate` object should
-be rendered and presented to the user as previously discussed.
-
-The uploader program must then verify `MetaUpdate.is_valid()` returns true.
-If not the uploader program must repeat the previous paragraph until it does.
-This is open ended and depends highly on properly formatting the `MetaData`
-object with well organized metadata.
-
-The uploader program should then call the `PolicyQuery.valid_metadata()` method
-to validate the completed metadata prior to upload. This prevents the uploader
-from uploading metadata not valid by the ingest policy. The uploader program
-should then parse the result for success then proceed.
-
-When the uploader program is ready to bundle the data. The uploader program
-must build an array of objects that represent the attributes of a `tar.TarInfo`
-object. Additionally, the object must have a `fileobj` attribute which is an
-object the bundler will call the `read()` method like a standard open file
-object.
-
-The uploader program must then create a `Bundler` object with the `MetaUpdate`
-and the array of `tar.TarInfo` objects as stated in the previous paragraph.
-The uploader program must then setup a writable file object and call the
-`stream()` method.
-
-The uploader program must then create a `Uploader` object. The uploader program
-must then setup a file-like object open for read and call the `upload()` method.
-
-The two file-like objects are open for read and write for upload and bundle
-respectively intentionally as this allows for setting up pipes between multiple
-processes to stream the data without keeping a local copy.
-
-The uploader must then verify the ingest has succeeded in ingest by calling
-`getstate()` and parsing the ingest response. If there is an error the uploader
-may perform the upload again.
+1. The uploader program MUST construct a new instance of the
+   `uploader.metadata.MetaUpdate` class. The new instance of the
+   `uploader.metadata.MetaUpdate` class MAY be associated with zero or more of
+   instances of the `uploader.metadata.MetaObj` class. The
+   `uploader.metadata.MetaObj.value` field MAY be `None`. The new instance of
+   the `uploader.metadata.MetaUpdate` class MUST NOT be associated with any
+   instances of the `uploader.metadata.FileObj` class.
+
+2. To determine completeness, the new instance of the
```

Comment:
There's two checks `MetaData.is_valid()` is a local check that should be executed first, the `PolicyQuery.valid_metadata()` is remote and should be done after the local check.